reframe/core/config.py

```diff
@@ -161,7 +161,10 @@ def _merge_config_sections(self, target, other):
             entry = functools.reduce(
                 lambda l, r: l.update(r) or l, optionset
             )
-            entry['target_systems'] = [system]
+            if 'target_systems' in entry.keys():
+                entry['target_systems'].append(system)
+            else:
+                entry['target_systems'] = [system]
```

Comment:
I would rewrite this more compactly:

```suggestion
            entry.setdefault('target_systems', [])
            entry['target_systems'].append(system)
```